README.md

```diff
@@ -1,69 +1,77 @@
-informing you of any potential errors related to TODO.
+Now, when you build your project, the Guice Checker will also run, informing you of any potential errors related to improperly defined or configured bindings.
 
 
 ## How to specify your code
 
-At compile time, the Dependency InjectionChecker estimates what values the program
-may compute at run time.  It issues a warning if the program may TODO.
-It works via a technique called pluggable typechecking.
+At compile time, the Guice Checker estimates what bindings the program may compute at run time.  It issues a warning if the program attempts to request a binding that has not been properly defined or configured. It works via a technique called pluggable typechecking.
 
-You need to specify the contracts of methods and fields in your code --
-that is, their requirements and their guarantees.  The Dependency InjectionChecker
-ensures that your code is consistent with the contracts, and that the
-contracts guarantee that TODO.
+Most of the time, the Guice Checker will work automatically. However, sometimes you may need to write an annotation to specify dataflow that is relevant to Guice bindings, if it crosses procedure boundaries. You can achieve that with the following type annotations:
 
-You specify your code by writing *qualifiers* such as `@DependencyInjectionBottom`
-on types, to indicate more precisely what values the type represents.
-Here are the type qualifiers that are supported by the Dependency InjectionChecker:
+`@Bind({})`
 
-`@DependencyInjectionUnknown`:
-The value might or might not be TODO. It is not safe to use for TODO.
-This is the default type, so programmers usually do not need to write it.
+- The default qualifier of this system. If an expression's type has this qualifier, then the expression is not a call to `com.google.inject.AbstractModule.bind` and it's value represents the fact that no arguments have been passed to the invocation of the `bind` method. As it is the default type, programmers usually do not need to write it.
 
-`@DependencyInjectionBottom`:
-The value is definitely TODO. It is safe to use for TODO.
+`@Bind({Baz.class})`
 
+- If an expression's type has this qualifier, then the expression is definitely a call to `com.google.inject.AbstractModule.bind` that received the argument `Baz.class` in an `AbstractModule`. It's value represents the argument that was passed to the invocation of the `bind` method.
 
-## How to build the checker
+`@BindAnnotatedWith({})`
 
-Run these commands from the top-level directory.
+- This qualifier cannot exist on its own. It is always a result of an `@Bind` and as such cannot be applied to an expression since it would mean that either a call to `bind` was made that received no arguments, _not possible_, or that no call to `bind` was ever made which would not be possible since `annotatedWith` is defined on the `AnnotatedBindingBuilder` class.
```

Comment:
I'd put this explanation after `@BAW` with arguments, since this explanation doesn't make sense until the reader knows what `@BAW` is for.